### PR TITLE
emit json serialization tags in go struct definitions; bump to 0.5.1

### PR DESCRIPTION
**Pull request by stevelr**
Signed-off-by: stevelr <steve@cosmonic.com>

### CODE REVIEW

**Comment by stevelr** [comment]
@jordan-rash if you build this with `cargo b` from source there'll be a binary file generated "target/debug/codegen" that you can run inside any folder that has `codegen.toml` to process the smithy files and generate code. Can you please test that this generates the right thing for your go project?

**Comment by jordan-rash** [comment]
> @jordan-rash if you build this with `cargo b` from source there'll be a binary file generated "target/debug/codegen" that you can run inside any folder that has `codegen.toml` to process the smithy files and generate code. Can you please test that this generates the right thing for your go project?

will do!

**Comment by stevelr** [comment]
regenerated interfaces with this build: https://github.com/wasmCloud/interfaces/pull/93

**Comment by stevelr** [comment]
This is in draft. Iterating with Jordan on go sdks

**Comment by jordan-rash** [comment]
Hey @stevelr ..... so this small commit fixes the imports the way I need them.  The only thing that isnt working, is specifically on the actor namespace, some of the functions are being generated with the incorrect prefix and I couldnt figure out how to fix that 

Generated:
```go
>> 30 type ActorSender struct { transport actor.Transport }
   29
   28                 // NewActorSender constructs a client for actor-to-actor messaging
   27                 // using the recipient actor's public key
   26                 func NewActorActorSender(actor_id string) *ActorSender {
>> 25                     transport := actor.ToActor(actor_id)
   24                     return &ActorSender { transport: transport }
   23                 }
   22
   21
>> 20 func (s *ActorSender) HealthRequest(ctx *actor.Context, arg actor.HealthCheckRequest) (*actor.HealthCheckResponse, error) {
   19
   18                     var sizer msgpack.Sizer
   17                     size_enc := &sizer
   16                     arg.MEncode(size_enc)
   15                     buf := make([]byte, sizer.Len())
   14
   13                     var encoder = msgpack.NewEncoder(buf)
   12                     enc := &encoder
   11                     arg.MEncode(enc)
   10
>>  9 out_buf,_ := s.transport.Send(ctx, actor.Message{ Method: "Actor.HealthRequest", Arg:buf })
    8 d := msgpack.NewDecoder(out_buf)
>>  7                         resp,err_ := actor.MDecodeHealthCheckResponse(&d)
    6                         if err_ != nil {
    5                             return nil,err_
    4                         }
    3                         return &resp,nil
    2                      }
```

Correct 
```go
>> 30 type ActorSender struct { transport Transport }
   29
   28                 // NewActorSender constructs a client for actor-to-actor messaging
   27                 // using the recipient actor's public key
   26                 func NewActorActorSender(actor_id string) *ActorSender {
>> 25                     transport := ToActor(actor_id)
   24                     return &ActorSender { transport: transport }
   23                 }
   22
   21
>> 20 func (s *ActorSender) HealthRequest(ctx *Context, arg core.HealthCheckRequest) (*core.HealthCheckResponse, error) {
   19
   18                     var sizer msgpack.Sizer
   17                     size_enc := &sizer
   16                     arg.MEncode(size_enc)
   15                     buf := make([]byte, sizer.Len())
   14
   13                     var encoder = msgpack.NewEncoder(buf)
   12                     enc := &encoder
   11                     arg.MEncode(enc)
   10
>>  9 out_buf,_ := s.transport.Send(ctx, Message{ Method: "Actor.HealthRequest", Arg:buf })
    8 d := msgpack.NewDecoder(out_buf)
>>  7                         resp,err_ := core.MDecodeHealthCheckResponse(&d)
    6                         if err_ != nil {
    5                             return nil,err_
    4                         }
    3                         return &resp,nil
    2                      }
```

File: codegen/src/codegen_go.rs [hunk 1]
```diff
@@ -345,13 +346,15 @@ impl<'model> CodeGen for GoCodeGen<'model> {
...
                 msgpack "github.com/wasmcloud/tinygo-msgpack" //nolint
-                cbor "github.com/wasmcloud/tinygo-cbor" //nolint
+                cbor "github.com/wasmcloud/tinygo-cbor"       //nolint
+                {}
             )"#,
             &self.package,
-            if ns != wasmcloud_model_namespace() && ns != wasmcloud_core_namespace() {
-                "\"github.com/wasmcloud/actor-tinygo\" //nolint"
+            if ns == wasmcloud_actor_namespace() {
+                "core \"github.com/wasmcloud/interfaces/core/tinygo\" //nolint"
```
Comment (stevelr) [inline codegen/src/codegen_go.rs:355]: @jordan-rash the two namespace changes are line 355 and 357

**Comment by stevelr** [comment]
merged into https://github.com/wasmCloud/weld/pull/120

**Pull request closed without merging**